app/assets/javascripts/forms.js

```diff
@@ -1,38 +1,107 @@
+// This file handles the behaviour for managing form building.
+//
+// Dependencies:
+//
+//   jQuery.ui.sortable: https://jqueryui.com/sortable/
+//   - Allows form fields to be dragged and re-oredered
+//
+//   Twitter's typeahead: http://twitter.github.io/typeahead.js/
+//   - Use for autocompleting for setting the field's name value
+
 (function(){
   var initialize = function(){
     makeSortable();
     bindHandlers();
+    autoComplete()
+  };
+
+  var whitelist = [
```

Comment:
No this whole file! It's ideal for encapsulating as an object. Making it a backbone view just makes you conform to a well-known code pattern. You describe the event bindings at the top as a hash, scope the DOM under one cached element, and make all the functions into methods. It's just about making the js more declarative and standardized. You can see an example from my new PR [here](https://github.com/SumOfUs/Champaign/pull/153/files#diff-a70738d77f62d44665e7196d2874e3d1R1)